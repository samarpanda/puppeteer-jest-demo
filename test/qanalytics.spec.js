const base = require('./../src/base');
const quikr = require('../src/qanalytics');
const URL = 'https://www.quikr.com';
const query = 'pet dog';
let browser;

beforeAll(async () => {
  browser = await base.start();
});

describe('Quikr search', () => {
  beforeEach(() => {
    jest.setTimeout(10000);
  });

  test('Get search input field', async done => {
    const page = await base.open(browser, URL);
    const result = await quikr.search(page, query);
    expect(result.length).toBe(2);
    done();
  });
});

afterAll(async () => {
  await base.stop(browser);
});
