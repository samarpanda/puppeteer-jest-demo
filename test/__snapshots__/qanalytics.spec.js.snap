// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quikr search Get search input field 1`] = `
Array [
  Object {
    "cookie_id": "7300a5c3-2136-4384-8f52-c09fa4b182e5",
    "device_info": Object {
      "browser_agent": "5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A372 Safari/604.1",
    },
    "events": Array [
      Object {
        "city_id": "1",
        "city_name": "www",
        "event_id": "6edf5f52-aa2c-42b1-9ad2-27d9ed7d2af8",
        "event_name": "Search",
        "event_time_epoc_ms": 1583990619717,
      },
    ],
    "page_info": Object {
      "ref_url": "",
      "url": "https://www.quikr.com/",
    },
    "property": "Quikr",
    "source": "Msite",
  },
  Object {
    "cookie_id": "7300a5c3-2136-4384-8f52-c09fa4b182e5",
    "device_info": Object {
      "browser_agent": "5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A372 Safari/604.1",
    },
    "events": Array [
      Object {
        "city": "India",
        "city_id": "1",
        "event_id": "6edf5f52-aa2c-42b1-9ad2-27d9ed7d2af8",
        "event_name": "Home_Page",
        "event_time_epoc_ms": 1583990619717,
      },
    ],
    "page_info": Object {
      "ref_url": "",
      "url": "https://www.quikr.com/",
    },
    "property": "Quikr",
    "source": "Msite",
  },
]
`;
